packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3047,12 +3067,49 @@ export function captureCommitPhaseError(sourceFiber: Fiber, error: mixed) {
           markRootUpdated(root, SyncLane, eventTime);
           ensureRootIsScheduled(root, eventTime);
           schedulePendingInteractions(root, SyncLane);
+        } else {
+          // This component has already been unmounted.
```

Comment:
I would fork the `captureCommitPhaseError` function so there's a separate one for errors in deleted trees. Then you can start the search from the parent instead of tracking this extra boolean state.

Could make it easier to fork the semantics if we do decide to go that route.

Reply:
Thinking about it while I was getting ready for my run– maybe it would be easier for us just to temporarily reconnect the disconnected `return` pointer before calling `captureCommitPhaseError`

Reply:
This directly almost works, in that it would bubble up to the nearest still-mounted boundary if there isn't a boundary in the unmounted tree– and it would schedule the new work (because it could bubble to the root)– _but_ it wouldn't call `componentDidCatch` for an error boundary still inside of the tree, since the work scheduling mechanism wouldn't traverse the previously unmounted subtree.

Will give it some more thought this morning.

Reply:
> I would fork the captureCommitPhaseError function so there's a separate one for errors in deleted trees. Then you can start the search from the parent instead of tracking this extra boolean state.

That would mean that we wouldn't call a logging error boundary if it was in the unmounted tree...which, maybe this is fine.